File: O(log n)/log_N.c
#include <stdio.h>

int buscaBinaria(int arr[], int tamanho, int alvo) {
    int inicio = 0;
    int fim = tamanho - 1;

    while (inicio <= fim) {
        int meio = inicio + (fim - inicio) / 2;   // Encontra o índice do meio

        if (arr[meio] == alvo)
            return meio; // Elemento encontrado

        if (arr[meio] < alvo)
            inicio = meio + 1; // Descarta a metade inferior
        else
            fim = meio - 1; //Descarta a metade superior
    }
    return -1; // Não encontrado
}

int main() {
    int numeros[] = {1, 3, 5, 7, 9, 11, 13};
    int tamanho = sizeof(numeros) / sizeof(numeros[0]);
    int alvo = 7;
    
    int resultado = buscaBinaria(numeros, tamanho, alvo);
    
    if (resultado != -1)
        printf("Elemento encontrado no indice: %d", resultado);
    else
        printf("Elemento nao encontrado");
        
    return 0;
}

File: O(log n)/log_n.py
def busca_binaria(lista, alvo):
    inicio = 0
    fim = len(lista) - 1

    while inicio <= fim:
        meio = (inicio + fim) // 2  # Encontra o índice do meio
        chute = lista[meio]

        if chute == alvo:
            return meio  # Encontrado
        if chute > alvo:
            fim = meio - 1  # Descarta a metade superior
        else:
            inicio = meio + 1  # Descarta a metade inferior
    
    return -1  # Não encontrado

# Teste
numeros = [1, 3, 5, 7, 9, 11, 13] # Lista deve estar ordenada
resultado = busca_binaria(numeros, 7)
print(f"Elemento encontrado no índice: {resultado}")